### PR TITLE
Issue #32: Add support for SSL certificate management.

### LINKED ISSUE #32: Add support for SSL management
The environment endpoints include support for managing SSL certificates. ([link](https://cloudapi-docs.acquia.com/#/); search for "ssl").

Please add some support for adding, removing, activating, and otherwise managing the SSL certificates in an environment.

### PR DESCRIPTION
**Pull request by jlfranklin**
Add support for the following endpoints:

* `get_ssl_settings()`: /environments/{environment_uuid}/ssl
* `get_ssl_certs()`: /environments/{environment_uuid}/ssl/certificates
* `get_ssl_cert()`: /environments/{environment_uuid}/ssl/certificates/{cert_id}
* `install_ssl_cert()`: /environments/{environment_uuid}/ssl/certificates (POST)
* `delete_ssl_cert()`: /environments/{environment_uuid}/ssl/certificates/{cert_id} (DELETE)
* `activate_ssl_cert()`: /environments/{environment_uuid}/ssl/certificates/{cert_id}/actions/activate (POST)
* `deactivate_ssl_cert()`: /environments/{environment_uuid}/ssl/certificates/{cert_id}/actions/deactivate (POST)

### CODE REVIEW

File: acapi2/resources/environment.py [hunk 1]
```diff
@@ -345,3 +345,97 @@ def enable_cron(self, cron_id: str) -> Session:
...
+    def get_ssl_certs(self) -> dict:
+        """
+        Return a list of SSL certificates.
+        """
+        uri = f"{self.uri}/ssl/certificates"
+        response = self.request(uri=uri)
+
+        return response.json().get('_embedded', {}).get('items')
+
+    def get_ssl_cert(self, cert_id) -> dict:
```
Comment (pmatias) [inline acapi2/resources/environment.py]: I think you forgot the type hint here for `cert_id` :)

File: acapi2/resources/environment.py [hunk 2]
```diff
@@ -345,3 +345,97 @@ def enable_cron(self, cron_id: str) -> Session:
...
+        }
+        if csr_id is not None:
+            data['csr_id'] = csr_id
+        if ca_certificates is not None:
+            data['ca_certificates'] = ca_certificates
+        response = self.request(uri=uri, method="POST", data=data)
+
+        return response
+
+    def delete_ssl_cert(self, cert_id) -> Session:
```
Comment (pmatias) [inline acapi2/resources/environment.py]: Same here.

File: acapi2/resources/environment.py [hunk 3]
```diff
@@ -345,3 +345,97 @@ def enable_cron(self, cron_id: str) -> Session:
...
+        """
+        Remove an SSL cert.
+        :param: cert_id: The Acquia certificate ID.
+        """
+        uri = f"{self.uri}/ssl/certificates/{cert_id}"
+        response = self.request(uri=uri, method="DELETE")
+
+        return response
+
+    def activate_ssl_cert(self, cert_id) -> Session:
```
Comment (pmatias) [inline acapi2/resources/environment.py]: And here.

File: acapi2/resources/environment.py [hunk 4]
```diff
@@ -345,3 +345,97 @@ def enable_cron(self, cron_id: str) -> Session:
...
+        """
+        Activate a previously installed SSL cert.
+        :param: cert_id: The Acquia certificate ID.
+        """
+        uri = f"{self.uri}/ssl/certificates/{cert_id}/actions/activate"
+        response = self.request(uri=uri, method="POST", data={})
+
+        return response
+
+    def deactivate_ssl_cert(self, cert_id) -> Session:
```
Comment (pmatias) [inline acapi2/resources/environment.py]: And also here :)

**Review by pmatias — CHANGES_REQUESTED** [review]
Hi @jlfranklin, thanks so much for your contrib!

Do you mind taking a look at the comments in the code, and, if possible, add the automated tests for the new methods? 

Thanks!

**Comment by jlfranklin** [comment]
Type-hinting I can do now.  Automated tests will have to wait until I have a block of time to devote to it.

**Pull request closed without merging**